cmd/attest.go

```diff
@@ -0,0 +1,179 @@
+package cmd
+
+import (
+	"fmt"
+	"io"
+	"strconv"
+
+	"cloud.google.com/go/compute/metadata"
+	"github.com/google/go-tpm-tools/client"
+	"github.com/google/go-tpm-tools/proto/attest"
+	"github.com/google/go-tpm/tpm2"
+	"github.com/spf13/cobra"
+	"google.golang.org/protobuf/proto"
+)
+
+var (
+	key      string
+	teeNonce []byte
+)
+
+var attestationKeys = map[string]map[tpm2.Algorithm]func(rw io.ReadWriter) (*client.Key, error){
+	"AK": {
+		tpm2.AlgRSA: client.AttestationKeyRSA,
+		tpm2.AlgECC: client.AttestationKeyECC,
+	},
+	"gceAK": {
+		tpm2.AlgRSA: client.GceAttestationKeyRSA,
+		tpm2.AlgECC: client.GceAttestationKeyECC,
+	},
+}
+
+var attestCmd = &cobra.Command{
+	Use:   "attest",
+	Short: "Create a remote attestation report",
+	Long: `Gather information for remote attestation.
+
+The Attestation report contains a quote on all available PCR banks, a way to validate 
+the quote, and a TCG Event Log (Linux only).
+
+Use --key to specify the type of attestation key. It can be gceAK for GCE attestation
+key or AK for a custom attestation key. By default it uses gceAK.
+--algo flag override the public key algorithm for attestation key. If not provided then
+by default rsa is used.
+--nonce attaches even length extra data to the report and guarantees a fresh quote.
+--teenonce attaches a 64 bytes extra data to the attestation report of hardware and 
+guarantees a fresh quote.
+--format flag specify the type of output file. It can be "binarypb" or "textproto".
+By default it uses binarypb.
+--output flag provides file name to store attestation report in the format specified
+by --format flag.
+`,
+	Args: cobra.NoArgs,
+	RunE: func(cmd *cobra.Command, args []string) error {
+
+		rwc, err := openTpm()
+		if err != nil {
+			return err
+		}
+		defer rwc.Close()
+
+		var attestationKey *client.Key
```

Comment:
We can simplify this as follows:
```
algoToAKCreate, ok := attestationKeys[key]
if !ok {
  return fmt.Errorf("key should be one of the following: %v", maps.Keys(attestationKeys))
}
createFunc := algoToAKCreate[keyAlgo]
attestationKey, err = createFunc(rwc)
...
```

Reply:
I have imported the maps package  and implemented it in the suggested way.

Reply:
The maps package is supported by go version 1.18 or more , so for now I'll add a hard coded error message saying `"key should be either gceAK or AK"`